sanic/signals.py

```diff
@@ -10,13 +10,39 @@
 from sanic_routing.utils import path_to_parts  # type: ignore
 
 from sanic.exceptions import InvalidSignal
+from sanic.log import error_logger, logger
 from sanic.models.handler_types import SignalHandler
 
 
-RESERVED_NAMESPACES = (
-    "server",
-    "http",
-)
+RESERVED_NAMESPACES = {
+    "server": (
+        # "server.main.start",
+        # "server.main.stop",
+        "server.init.before",
+        "server.init.after",
+        "server.shutdown.before",
+        "server.shutdown.after",
+    ),
+    "http": (
+        "http.lifecycle.begin",
+        "http.lifecycle.complete",
+        "http.lifecycle.exception",
+        "http.lifecycle.handle",
+        "http.lifecycle.read_body",
+        "http.lifecycle.read_head",
+        "http.lifecycle.request",
+        "http.lifecycle.response",
+        "http.routing.after",
+        "http.routing.before",
+        "http.lifecycle.send",
+        "http.middleware.after",
+        "http.middleware.before",
+    ),
+}
```

Comment:
future thought: this could be a tree, perhaps? I mean, even I'm not sure if that's a good idea, hence the "future thought" ...

Reply:
It was originally, and then I realized I was over-complicating it. I think this simple structure (1) does what we need, and (2) is super simple to whoever is looking at the code.

Reply:
yeah, then +1 to my attempt of over-complicate things :laughing: